### PR TITLE
DTSPO-14865-replace-kured-v2

### PR DESCRIPTION
**Pull request by JoshuaGlassKainos**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/DTSPO-14865

### Change description ###
added maintenance window config obj

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: components/aks/inputs-default.tf [hunk 1]
```diff
@@ -41,4 +41,25 @@ variable "autoShutdown" {
 
 variable "startupMode" {
   default = null
+}
+
+variable "node_os_maintenance_window_config" {
+  type = object({
+    frequency   = string
+    interval    = number
```
Comment (thomast1906) [inline components/aks/inputs-default.tf:49]: Do we want a default as below or default values if not entered?

Currently your sbox is somewhat duplicating the defaults _(just a thought)_

Comment (JoshuaGlassKainos) [reply components/aks/inputs-default.tf:49]: It's more for it defaults if no `node_os_maintenance_window_config` is passed through. could remove it from here as defaults are given in the module.

Comment (JoshuaGlassKainos) [reply components/aks/inputs-default.tf:49]: We can't have it bundle up missing values as this will mess with validation down the line

Comment (JoshuaGlassKainos) [reply components/aks/inputs-default.tf:49]: There was a possibility to do the following
```
  type = object({
    frequency   = (string, "Weekly")
    interval    = number
    duration    = number
    day_of_week = optional(string, "Monday")
    start_time  = optional(string, "18:00")
    utc_offset  = optional(string, "+00:00")
    start_date  = optional(string, null)
  })
```

and the inputs would just have to be 

```
node_os_maintenance_window_config = {
  interval    = 1
  duration    = 4
}
```
which would output the following, but if a user enters `frequency = Daily` then they would also have to null `day_of_week` as both can't be inputed at the same time
```
maintenance_window_node_os {
  frequency   = "Weekly"
  interval    = 1
  duration    = 4
  day_of_week = "Monday"
  start_time  = "16:00"
  utc_offset  = "+00:00"
}
```

**Comment by JoshuaGlassKainos** [comment]
will re run tests once module has been updated

**Review by danielwilsonkainos — APPROVED** [review]

**Pull request merged**